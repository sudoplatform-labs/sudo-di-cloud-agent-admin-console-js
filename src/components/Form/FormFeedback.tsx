import React from 'react';
import styled from 'styled-components';

interface FormFeedbackProps {
  id?: string;
  showText: boolean;
  className?: string;
  children: React.ReactNode;
}

const FormFeedbackComponent = styled.div<FormFeedbackProps>`
  overflow: hidden;
  height: auto;
  max-height: ${(props) => (props.showText ? '600px' : '0')};
  transition: max-height 0.25s ease-in-out;
`;

export const FormFeedback = (props: FormFeedbackProps): React.ReactElement => (
  <FormFeedbackComponent showText={props.showText} className={props.className}>
    {props.children}
  </FormFeedbackComponent>
);
