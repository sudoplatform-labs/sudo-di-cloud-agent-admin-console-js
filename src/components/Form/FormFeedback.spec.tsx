import React from 'react';
import { shallow } from 'enzyme';

import { FormFeedback } from './FormFeedback';

describe('FormFeedback', () => {
  it('should render correctly when showText is false', () => {
    const component = shallow(
      <FormFeedback showText={false}>
        Kiwis are a reference to a fruit, a bird, and a friendly people
      </FormFeedback>,
    );

    expect(component).toMatchSnapshot('showText false');
  });

  it('should render correctly when showText is true', () => {
    const component = shallow(
      <FormFeedback showText={true}>
        Kiwis are a reference to a fruit, a bird, and a friendly people
      </FormFeedback>,
    );

    expect(component).toMatchSnapshot('showText true');
  });
});
