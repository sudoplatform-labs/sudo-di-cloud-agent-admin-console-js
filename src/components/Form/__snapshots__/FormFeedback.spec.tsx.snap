// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormFeedback should render correctly when showText is false: showText false 1`] = `
<styled.div
  showText={false}
>
  Kiwis are a reference to a fruit, a bird, and a friendly people
</styled.div>
`;

exports[`FormFeedback should render correctly when showText is true: showText true 1`] = `
<styled.div
  showText={true}
>
  Kiwis are a reference to a fruit, a bird, and a friendly people
</styled.div>
`;
